### PR TITLE
Fix resize case, currently resize case always fail for no capability.

### PR DESCRIPTION
**Pull request by LiliDeng**

### CODE REVIEW

File: lisa/sut_orchestrator/azure/platform_.py [hunk 1]
```diff
@@ -1573,6 +1573,7 @@ def _resource_sku_to_capability(  # noqa: C901
                 schema.FeatureSettings.create(features.StartStop.name()),
                 schema.FeatureSettings.create(features.SerialConsole.name()),
                 schema.FeatureSettings.create(features.Hibernation.name()),
+                schema.FeatureSettings.create(features.Resize.name()),
```
Comment (squirrelsc) [inline lisa/sut_orchestrator/azure/platform_.py:1593]: My understanding is not all vm size can be resize. Is there a list of it?

Comment (LiliDeng) [reply lisa/sut_orchestrator/azure/platform_.py:1593]: probably no, but when resize a vm, I know there is an api to know the vm sizes list which can be resized based on current vm size. if no sizes listed, then can skip the case, I will check.

File: microsoft/testsuites/core/vm_hot_resize.py [hunk 2]
```diff
@@ -37,14 +39,30 @@ class VmHotResize(TestSuite):
...
+                expected_vm_capability = None
+                expected_vm_capability = resize.resize()
+                break
+            except Exception as e:
+                if "cannot find current vm size in eligible list" in str(e):
+                    retry = retry + 1
+                elif "OperationNotAllowed" in str(e):
+                    retry = retry + 1
+                else:
+                    log.debug("unexpected exception found")
```
Comment (squirrelsc) [inline microsoft/testsuites/core/vm_hot_resize.py]: For unknown exception, use `raise identifier` to reraise it.

File: microsoft/testsuites/core/vm_hot_resize.py [hunk 3]
```diff
@@ -37,14 +39,30 @@ class VmHotResize(TestSuite):
...
         resize = node.features[Resize]
-        expected_vm_capability = resize.resize()
+        retry = 1
+        maxretry = 10
+        while retry < maxretry:
+            try:
+                expected_vm_capability = None
+                expected_vm_capability = resize.resize()
+                break
+            except Exception as e:
```
Comment (squirrelsc) [inline microsoft/testsuites/core/vm_hot_resize.py]: In other places, we use `identifier` as the name.

File: microsoft/testsuites/core/vm_hot_resize.py [hunk 4]
```diff
@@ -37,14 +39,30 @@ class VmHotResize(TestSuite):
...
     )
-    def verify_vm_hot_resize(self, node: Node) -> None:
+    def verify_vm_hot_resize(self, log: Logger, node: Node) -> None:
         resize = node.features[Resize]
-        expected_vm_capability = resize.resize()
+        retry = 1
+        maxretry = 10
+        while retry < maxretry:
+            try:
+                expected_vm_capability = None
```
Comment (squirrelsc) [inline microsoft/testsuites/core/vm_hot_resize.py]: It's not easy to understand.

```python
expected_vm_capability: Optional[schema.NodeSpace] = None
try:
    expected_vm_capability = resize.resize()
```

File: microsoft/testsuites/core/vm_hot_resize.py (hunk 2 above)
Comment (LiliDeng) [reply microsoft/testsuites/core/vm_hot_resize.py]: update

File: microsoft/testsuites/core/vm_hot_resize.py (hunk 3 above)
Comment (LiliDeng) [reply microsoft/testsuites/core/vm_hot_resize.py]: update

File: microsoft/testsuites/core/vm_hot_resize.py (hunk 4 above)
Comment (LiliDeng) [reply microsoft/testsuites/core/vm_hot_resize.py]: update

**Review by squirrelsc — APPROVED** [review]

**Pull request merged**